packages/projection-typeorm/test/operators/storeAddresses.test.ts

```diff
@@ -0,0 +1,196 @@
+    addressesRepo = queryRunner.manager.getRepository(AddressEntity);
+    buffer = new TypeormStabilityWindowBuffer({ allowNonSequentialBlockHeights: true, logger });
+    await buffer.initialize(queryRunner);
+  });
+
+  afterEach(async () => {
+    await queryRunner.release();
+    buffer.shutdown();
+  });
+
+  it('inserts addresses with their type, payment credential and staking credential', async () => {
+    const { addresses } = await projectTilFirst((evt) => evt.addresses.some(isAddressWithBothCredentials));
+    const { address, type, paymentCredentialHash, stakingCredential } = addresses.find(isAddressWithBothCredentials)!;
+    const projectedAddress = await addressesRepo.findOne({ where: { address } })!;
+    expect(projectedAddress).toEqual({
+      address,
+      paymentCredentialHash,
+      stakingCredentialHash: stakingCredential,
+      type
+    });
+  });
+
+  describe('pointer addresses', () => {
+    // eslint-disable-next-line unicorn/consistent-function-scoping
+    const projectEvtWithAddress = (address: Cardano.PaymentAddress, blockNo: Cardano.BlockNo) =>
+      firstValueFrom(
+        // clone the same block, just bump/patch the header
+        createStubProjectionSource([
+          createStubRollForwardEvent(
+            {
+              blockBody: [
+                {
+                  body: {
+                    fee: 123n,
+                    inputs: [],
+                    outputs: [
+                      {
+                        address,
+                        value: {
+                          coins: 123n
+                        }
+                      }
+                    ]
+                  },
+                  id: Cardano.TransactionId(generateRandomHexString(64)),
+                  inputSource: Cardano.InputSource.inputs,
+                  witness: { signatures: new Map() }
+                }
+              ],
+              blockHeader: createStubBlockHeader(blockNo)
+            },
+            stubEvents.networkInfo
+          )
+        ]).pipe(applyOperators)
+      );
+
+    it('looks up stake key hash by pointer', async () => {
+      const stakeKeyRegistrationEvent = await projectTilFirst((evt) => evt.stakeKeyRegistrations.length > 0);
+      const registration = stakeKeyRegistrationEvent.stakeKeyRegistrations[0];
+      const address = Cardano.PointerAddress.fromCredentials(
+        Cardano.NetworkId.Testnet,
+        cip19TestVectors.KEY_PAYMENT_CREDENTIAL,
+        registration.pointer
+      )
+        .toAddress()
+        .toBech32() as Cardano.PaymentAddress;
+      await projectEvtWithAddress(address, Cardano.BlockNo(stakeKeyRegistrationEvent.block.header.blockNo + 1));
+      const projectedAddress = await addressesRepo.findOne({ where: { address } });
+      expect(typeof projectedAddress!.stakingCredentialHash).toBe('string');
+    });
+
+    it('projects invalid pointers (pointing to nothing) as undefined staking credential', async () => {
```

Comment:
```suggestion
    it('projects null pointers as undefined staking credential', async () => {
```

Reply:
27e2ff443fd191109bdcda6b9d1595866170e893 changed the suggested message to 'stake credential'